frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -2542,9 +2542,14 @@ class ResolvedExpression {
   /** set the toId */
   void setToId(ID toId) { toId_ = toId; }
 
-  /** set the type */
+  /** set the qualified type */
   void setType(const types::QualifiedType& type) { type_ = type; }
 
+  /** set the kind of the qualified type */
+  void setKind(types::QualifiedType::Kind kind) {
+    type_ = { kind, type_.type(), type_.param() };
```

Comment:
nit: a bit scary because this can result in something that's non-`PARAM` but has a param pointer, and the opposite way.

Reply:
How about I add an assertion ensuring that the kind and param-pointer are consistent?